sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -2623,8 +2628,16 @@ object Sequence {
         // about a month length in days and a day length in microseconds
         val intervalStepInMicros =
           stepMicros + stepMonths * microsPerMonth + stepDays * microsPerDay
-        val startMicros: Long = num.toLong(start) * scale
-        val stopMicros: Long = num.toLong(stop) * scale
+
+        // Date to timestamp is not equal from GMT and Chicago timezones
```

Comment:
can we exclude these changes and make this PR smaller to only focus on "forbid time field steps for date start/end"?

Reply:
@cloud-fan Thanks, I have followed the suggestion and make a new jira ticket for this.